src/countdown-timer/countdown-timer.tsx

```diff
@@ -0,0 +1,154 @@
+import { useEffect, useRef, useState } from "react";
+import { useMediaQuery } from "react-responsive";
+import { useTimer } from "./use-timer";
+import { CountdownTimerProps } from "./types";
+import { Countdown, Time, TimeLeft, Wrapper } from "./countdown-timer.style";
+import { TimeHelper } from "../util/time-helper";
+import { ClockIcon } from "@lifesg/react-icons";
+import { useIntersectionObserver } from "../util/use-intersection-observer";
+import { MediaWidths } from "../spec/media-spec";
+
+type Position = "relative" | "fixed";
+
+export const CountdownTimer = ({
+    className,
+    timer: _timer,
+    notifyTimer,
+    offset,
+    mobileOffset,
+    show,
+    "data-testid": testId,
+    onDuration,
+    onNotify,
+    onFinish,
+}: CountdownTimerProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+
+    const wrapperRef = useRef<HTMLDivElement>();
+    const stickyRef = useRef<HTMLDivElement>();
+    const isNotified = useRef<boolean>(false);
+    const [offsetY, setOffsetY] = useState<number>(0);
+
+    const [timer, isPlaying, setIsPlaying] = useTimer(_timer);
+    const onScreen = useIntersectionObserver(stickyRef, {
+        threshold: 1,
+        rootMargin: `${offsetY * -1}px 0px 0px 0px`,
+    });
+
+    const isMobile = useMediaQuery({
+        maxWidth: MediaWidths.mobileL,
+    });
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+
+    useEffect(() => {
+        setIsPlaying(show);
+    }, [show]);
+
+    useEffect(() => {
+        if (timer === 0) {
+            performOnFinishHandler();
+        } else if (timer === notifyTimer) {
+            performOnDurationHandler();
+            performOnNotifyHandler();
+        } else if (timer < notifyTimer) {
+            if (!isNotified.current) performOnNotifyHandler();
+            performOnDurationHandler();
+        }
+    }, [timer]);
+
+    useEffect(() => {
+        const y = getOffsetY();
+        setOffsetY(y);
+    }, [isMobile, wrapperRef.current]);
```

Comment:
initial the wrapperRef is null if `show` is false. It getting the offsetY once `show` is `true`. 

It able to remove right now since we request to track `offset?.top` inside useEffect.